### PR TITLE
build(routerlicious-urlresolver): use fluid-tsc -- nconf

### PR DESCRIPTION
**Pull request by DLehenbauer**
[AB#7353](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7353)

nconf import problem:
```
file:///workspaces/f1/packages/drivers/routerlicious-urlResolver/lib/test/routerlicious-urlResolver.spec.js:6
import { Provider } from "nconf";
         ^^^^^^^^
SyntaxError: Named export 'Provider' not found. The requested module 'nconf' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from 'nconf';
const { Provider } = pkg;
```

Replace `nconf` with `nconf-esm` which is fully ESM compatible: https://www.npmjs.com/package/nconf-esm

### CODE REVIEW

**Review by RishhiB — APPROVED** [review]

File: packages/drivers/routerlicious-urlResolver/package.json [hunk 1]
```diff
@@ -88,9 +70,11 @@
 		"prettier": "prettier --check . --cache --ignore-path ../../../.prettierignore",
 		"prettier:fix": "prettier --write . --cache --ignore-path ../../../.prettierignore",
 		"test": "npm run test:mocha",
-		"test:mocha": "mocha --ignore \"dist/test/types/*\" --recursive dist/test -r node_modules/@fluid-internal/mocha-test-setup",
+		"test:mocha": "npm run test:mocha:esm && echo skipping cjs to avoid overhead - npm run test:mocha:cjs",
```
Comment (jason-ha) [inline packages/drivers/routerlicious-urlResolver/package.json:73]: This is using a package that is "special" compared to prior CJS version. Hopefully it handles both, but we should test both.

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -7960,17 +7960,16 @@ importers:
...
       rimraf: ^4.4.0
-      tsc-multi: ^1.1.0
       typescript: ~5.1.6
     dependencies:
       '@fluidframework/core-interfaces': link:../../common/core-interfaces
       '@fluidframework/core-utils': link:../../common/core-utils
       '@fluidframework/driver-definitions': link:../../common/driver-definitions
       '@fluidframework/protocol-definitions': 3.2.0-237840
-      nconf: 0.12.1
+      nconf-esm: 0.0.6
```
Comment (jason-ha) [inline pnpm-lock.yaml:7972]: I am not seeing `nconf` completely disappear from the package group dependencies. Where else is it being used and is it a problem? Or why not?

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: packages/drivers/routerlicious-urlResolver/src/urlResolver.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ const r11sServers = [
 export class RouterliciousUrlResolver implements IUrlResolver {
 	constructor(
 		private readonly config:
-			| { provider: Provider; tenantId: string; documentId: string }
+			| { provider: nconf.Provider; tenantId: string; documentId: string }
```
[the remark this answers is not in the document]
Comment (jason-ha) [reply packages/drivers/routerlicious-urlResolver/src/urlResolver.ts:24]: Looks like test-version-utils should be updated to confirm this pattern suffices.

File: pnpm-lock.yaml (hunk 2 above)
Comment (jason-ha) [reply pnpm-lock.yaml]: Looks like we are keeping nconf now; so, resolving this.

Comment (sonalideshpandemsft) [reply pnpm-lock.yaml]: Good point. The distinction lies in how it's imported. In this case, we import the default export from the nconf module and then extract the Provider object from it.

File: packages/drivers/routerlicious-urlResolver/package.json (hunk 1 above)
Comment (sonalideshpandemsft) [reply packages/drivers/routerlicious-urlResolver/package.json]: why is this special?

File: packages/drivers/routerlicious-urlResolver/src/urlResolver.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (sonalideshpandemsft) [reply packages/drivers/routerlicious-urlResolver/src/urlResolver.ts]: test-version-utils is not happy...added `nconf.cts` file as you recommended @jason-ha :)

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: packages/drivers/routerlicious-urlResolver/src/nconf.cts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+ * Licensed under the MIT License.
+ */
+
+// This file serves as an intermediary module to export the 'Provider' named export from the 'nconf' module.
+// It is written using ESM module syntax, making it compatible with both ESM and CJS module.
+// This compatibility allows for flexible usage of the 'Provider' named export from the 'nconf' module across different module systems without modification.
+
+import nconf, { Provider } from "nconf";
+
+export { nconf, Provider };
```
[the remark this answers is not in the document]
Comment (sonalideshpandemsft) [reply packages/drivers/routerlicious-urlResolver/src/nconf.cts]: nconf is not required. removed

**Review by jason-ha — APPROVED** [review]

**Pull request merged**